From: Cliff Wickman <cpw@sgi.com>
Subject: x86/UV2: Remove stale no-resources test for UV2 BAU
Git-commit: 478c6e529e7bd7c6ef8994c55bd252c287c35893
Patch-mainline: Yes
References: bnc#743212

This patch removes an unnecessary test for a no-destination-resources-available
condition that looks like a destination timeout in UV1, but is separately
distinguishable in UV2.

Signed-off-by: Cliff Wickman <cpw@sgi.com>
Signed-off-by: Rafael J. Wysocki <rjw@suse.de>
---
 arch/x86/platform/uv/tlb_uv.c |   10 ----------
 1 file changed, 10 deletions(-)

Index: linux-3.0-SLE11-SP2/arch/x86/platform/uv/tlb_uv.c
===================================================================
--- linux-3.0-SLE11-SP2.orig/arch/x86/platform/uv/tlb_uv.c
+++ linux-3.0-SLE11-SP2/arch/x86/platform/uv/tlb_uv.c
@@ -642,16 +642,6 @@ static int uv2_wait_completion(struct ba
 		} else if (descriptor_stat == UV2H_DESC_DEST_TIMEOUT) {
 			stat->s_dtimeout++;
 			ttm = get_cycles();
-			/*
-			 * Our retries may be blocked by all destination
-			 * swack resources being consumed, and a timeout
-			 * pending.  In that case hardware returns the
-			 * ERROR that looks like a destination timeout.
-			 */
-			if (cycles_2_us(ttm - bcp->send_message) < timeout_us) {
-				bcp->conseccompletes = 0;
-				return FLUSH_RETRY_PLUGGED;
-			}
 			bcp->conseccompletes = 0;
 			return FLUSH_RETRY_TIMEOUT;
 		} else {
